include/cynq/accelerator.hpp

```diff
@@ -0,0 +1,74 @@
+/*
+ * See LICENSE for more information about licensing
+ *
+ * Copyright 2023
+ * Author: Luis G. Leon-Vega <luis.leon@ieee.org>
+ *         Diego Arturo Avila Torres <diego.avila@uned.cr>
+ *
+ */
+#pragma once
+#include "cynq/enums.hpp"
+#include "cynq/status.hpp"
+
+namespace cynq {
+class IAccelerator {
+ public:
+  virtual ~IAccelerator() {}
```

Comment:
you can use `virtual ~IAccelerator() = default;` instead.

Also, document it. Apply to all occurrences